contracts/src/node-interface/NodeInterface.sol

```diff
@@ -64,4 +64,31 @@ interface NodeInterface {
      * @return confirmations The number of L1 confirmations the sequencer batch has. Returns 0 if block not yet included in an L1 batch.
      */
     function getL1Confirmations(bytes32 blockHash) external view returns (uint64 confirmations);
+
+    /**
+     * @notice Same as native gas estimation, but with additional info on the l1 costs
```

Comment:
We should note for each of these methods if you're supposed to `eth_call` them or `eth_estimateGas` them. It might be confusing to end users that you need to call estimateGas for `estimateRetryableTicket` but call for `gasEstimateComponents`.

Reply:
could we have an `eth_call` on `estimateRetryableTicket` trigger the estimateGas behaviour? (ie do the binary search)
then we can have all methods in NodeInterface expect an `eth_call`

Reply:
I've added some comments to indicate which rpc to use. Having `eth_call` translate to `eth_estimateGas` is possible but requires some geth changes that include special casing so let's avoid if it possible